sdk/storage/azblob/service/client_test.go

```diff
@@ -709,6 +714,67 @@ func (s *ServiceUnrecordedTestsSuite) TestSASServiceClient() {
 	_require.Nil(err)
 }
 
+func (s *ServiceRecordedTestsSuite) TestSASServiceClientNoKey() {
+	_require := require.New(s.T())
+	accountName := os.Getenv("AZURE_STORAGE_ACCOUNT_NAME")
+
+	serviceClient, err := service.NewClientWithNoCredential(fmt.Sprintf("https://%s.blob.core.windows.net/", accountName), nil)
+	_require.Nil(err)
+	resources := sas.AccountResourceTypes{
+		Object:    true,
+		Service:   true,
+		Container: true,
+	}
+	permissions := sas.AccountPermissions{
+		Read:                  true,
+		Write:                 true,
+		Delete:                true,
+		DeletePreviousVersion: true,
+		List:                  true,
+		Add:                   true,
+		Create:                true,
+		Update:                true,
+		Process:               true,
+		Tag:                   true,
+		FilterByTags:          true,
+		PermanentDelete:       true,
+	}
+
+	expiry := time.Now().Add(time.Hour)
+	_, err = serviceClient.GetSASURL(resources, permissions, expiry, nil)
+	_require.Error(err, "SAS can only be signed with a SharedKeyCredential")
```

Comment:
Same as previous comment. Convert this to `_require.Equal`.

Reply:
this should be fixed now